scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala

```diff
@@ -471,4 +471,12 @@ class SyntacticSuite extends scala.meta.tests.parsers.ParseSuite {
     assert(t"((a, b)) => c".syntax    == "((a, b)) => c")
     assert(t"((a, b), c) => c".syntax == "((a, b), c) => c")
   }
+
+  test("Term.Apply(_, Seq(Term.Function(...))) #572") {
+    import scala.collection.immutable.Seq
+    val tree1= Term.Apply(Term.Name("foo"), Seq(Term.Function(Seq(Term.Param(Seq(), Term.Name("i"), Some(Type.Name("Int")), None)), Lit(()))))
+    val tree2= Term.Apply(Term.Name("foo"), Seq(Term.Function(Seq(Term.Param(Seq(Mod.Implicit()), Term.Name("i"), Some(Type.Name("Int")), None)), Lit(()))))
+    assert(tree1.toString == "foo { (i: Int) => () }")
```

Comment:
- Any reason to use `Term.Apply(...)` instead of quasiquotes? It's the same, but I think quasiquotes are more readable. 
- also, `.syntax` instead of `.toString`